tests/antithesis/test-template/robustness/traffic/main.go

```diff
@@ -137,58 +138,119 @@ func runTraffic(ctx context.Context, lg *zap.Logger, tf traffic.Traffic, hosts [
 	var mux sync.Mutex
 	var wg sync.WaitGroup
-	storage := identity.NewLeaseIDStorage()
-	limiter := rate.NewLimiter(rate.Limit(profile.MaximalQPS), profile.BurstableQPS)
-	concurrencyLimiter := traffic.NewConcurrencyLimiter(profile.MaxNonUniqueRequestConcurrency)
-	finish := closeAfter(ctx, duration)
 	reports := []report.ClientReport{}
-	keyStore := traffic.NewKeyStore(10, "key")
-	for i := 0; i < profile.ClientCount; i++ {
-		c := connect([]string{hosts[i%len(hosts)]}, ids, baseTime)
-		wg.Add(1)
-		go func(c *client.RecordingClient) {
-			defer wg.Done()
-			defer c.Close()
-
-			tf.RunTrafficLoop(ctx, c, limiter,
-				ids,
-				storage,
-				concurrencyLimiter,
-				keyStore,
-				finish,
-			)
-			mux.Lock()
-			reports = append(reports, c.Report())
-			mux.Unlock()
-		}(c)
-	}
-	wg.Add(1)
-	compactClient := connect(hosts, ids, baseTime)
-	go func(c *client.RecordingClient) {
-		defer wg.Done()
-		defer c.Close()
-		tf.RunCompactLoop(ctx, c, traffic.DefaultCompactionPeriod, finish)
-		mux.Lock()
-		reports = append(reports, c.Report())
-		mux.Unlock()
-	}(compactClient)
-	defragPeriod := traffic.DefaultCompactionPeriod * time.Duration(len(hosts))
-	for _, h := range hosts {
-		c := connect([]string{h}, ids, baseTime)
-		wg.Add(1)
-		go func(c *client.RecordingClient) {
-			defer wg.Done()
-			defer c.Close()
-			runDefragLoop(ctx, c, defragPeriod, finish)
-			mux.Lock()
-			reports = append(reports, c.Report())
-			mux.Unlock()
-		}(c)
+	sim := &simulator{
+		hosts:              hosts,
+		tf:                 tf,
+		reporting:          reporter{&wg, &mux, &reports},
+		ids:                ids,
+		baseTime:           baseTime,
+		finish:             closeAfter(ctx, duration),
+		storage:            identity.NewLeaseIDStorage(),
+		limiter:            rate.NewLimiter(rate.Limit(profile.MaximalQPS), profile.BurstableQPS),
+		concurrencyLimiter: traffic.NewConcurrencyLimiter(profile.MaxNonUniqueRequestConcurrency),
+		keyStore:           traffic.NewKeyStore(10, "key"),
 	}
+
+	sim.runSingleMemberTraffic(ctx)
+	sim.runAllMemberTraffic(ctx)
+	sim.runCompaction(ctx)
+	sim.runDefragment(ctx)
 	wg.Wait()
 	return reports
 }
 
+type simulator struct {
+	hosts              []string
+	tf                 traffic.Traffic
+	reporting          reporter
+	ids                identity.Provider
+	baseTime           time.Time
+	finish             <-chan struct{}
+	storage            identity.LeaseIDStorage
+	limiter            *rate.Limiter
+	concurrencyLimiter traffic.ConcurrencyLimiter
+	keyStore           *traffic.KeyStore
+}
+
+func (s *simulator) runSingleMemberTraffic(ctx context.Context) {
+	for i := range profile.MemberClientCount {
+		c := connect([]string{s.hosts[i%len(s.hosts)]}, s.ids, s.baseTime)
+		go s.reporting.makeReportOf(trafficLoad(ctx, s.tf, s.limiter, s.ids, s.storage, s.concurrencyLimiter, s.keyStore, s.finish)).usingClient(c)
+	}
+}
+
+func (s *simulator) runAllMemberTraffic(ctx context.Context) {
```

Comment:
There is no reason to have separate function for running all member traffic vs single member traffic. It's almost the same code structure. Separating things like this implies that they are less related causing potential confusion in reader.